app/src/main/java/com/example/weather/MainActivity.java

```diff
@@ -82,17 +82,18 @@ private void onClickByCity(View view) {
             public void run() {
                 Log.d(TAG, "run started");
 
+                WeatherInfo weather;
                 String respond;
-                String temperature;
+
                 try {
                     respond = doRequest(request);
                     if (TextUtils.isEmpty(respond)) {
                         notificationOnError(getText(R.string.error_wrong_request).toString());
                         return;
                     }
 
-                    temperature = parseTemperature(respond);
-                    if (TextUtils.isEmpty(temperature)) {
+                    weather = parseWeather(respond);
+                    if (weather.isEmpty) {
```

Comment:
weather info should not have property 'isEmpty' since there is no way that we receive empty weather info.
you can use Optional for this purpose.